datastore/modelassertmanager.go

```diff
@@ -0,0 +1,154 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package datastore
+
+import (
+	"errors"
+	"log"
+	"time"
+)
+
+const createModelAssertTableSQL = `
+	CREATE TABLE IF NOT EXISTS modelassertion (
+		id               serial primary key not null,
+		model_id         int references model not null,
+		keypair_id       int references keypair not null,
+		series           int not null,
+		architecture     varchar(20) not null,
+		revision         int not null default 0,
+		gadget           varchar(60) not null,
+		kernel           varchar(60) not null,
+		store            varchar(60),
+		created          timestamp default current_timestamp,
+		modified         timestamp default current_timestamp
+	)
+`
+const createModelAssertSQL = "INSERT INTO modelassertion (model_id,keypair_id,series,architecture,revision,gadget,kernel,store) VALUES ($1,$2,$3,$4,$5,$6,$7,$8) RETURNING id"
+
+const updateModelAssertSQL = `
```

Comment:
Cosmetic thing: Why some sql is padded and other not?

Reply:
Just the longer ones tend to be padded. The INSERT statements are usually shorter.